.github/workflows/tnf-image.yaml

```diff
@@ -0,0 +1,128 @@
+        id: set_tnf_version
+        env:
+          VERSION_FROM_FILE: ${{ fromJSON(steps.get_version_json_file.outputs.json)['tag'] }}
+
+  test-and-push-tnf-image:
+    name: 'Test and push the `test-network-function` image'
+    needs: [get-latest-tnf-version-number]
+    runs-on: ubuntu-20.04
+    env:
+      SHELL: /bin/bash
+      KUBECONFIG: '/home/runner/.kube/config'
+      TNF_VERSION: ${{ needs['get-latest-tnf-version-number'].outputs.TNF_VERSION }}
+
+    steps:
+      - name: Ensure $TNF_VERSION and $IMAGE_TAG are set
+        run: '[[ -n "$TNF_VERSION" ]] && [[ -n "$IMAGE_TAG" ]]'
+
+      - name: Check whether the version tag exists on remote
+        run: git ls-remote --exit-code $TNF_SRC_URL refs/tags/$TNF_VERSION
+
+      - name: (if tag is missing) Display debug message
+        if: ${{ failure() }}
+        run: echo "Tag '$TNF_VERSION' does not exist on remote $TNF_SRC_URL"
+
+      - name: Checkout the version tag
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ env.TNF_VERSION }}
+
+      - name: Build the `test-network-function` image
+        run: |
+          docker build --no-cache \
+            -t ${IMAGE_NAME}:${IMAGE_TAG} \
+            -t ${REGISTRY}/${IMAGE_NAME}:${IMAGE_TAG} \
+            -t ${REGISTRY}/${IMAGE_NAME}:${TNF_VERSION} \
+            --build-arg TNF_VERSION=${TNF_VERSION} \
+            --build-arg TNF_SRC_URL=${TNF_SRC_URL} .
+
+      # Create a minikube cluster for testing.
+
+      - name: Check out `cnf-certification-test-partner`
+        uses: actions/checkout@v2
+        with:
+          repository: test-network-function/cnf-certification-test-partner
+          path: cnf-certification-test-partner
+
+      - name: Start the minikube cluster for `local-test-infra`
+        uses: ./cnf-certification-test-partner/.github/actions/start-minikube
+
+      - name: Create `local-test-infra` OpenShift resources
+        uses: ./cnf-certification-test-partner/.github/actions/create-local-test-infra-resources
+        with:
+          working_directory: cnf-certification-test-partner
+
+      # Perform tests.
+
+      - name: Create required TNF config files and directories
+        run: |
+          mkdir -p $TNF_CONFIG_DIR $TNF_OUTPUT_DIR
+          cp test-network-function/*.yml $TNF_CONFIG_DIR
+        shell: bash
+
+      - name: 'Test: Run diagnostic test suite'
+        run: ./run-container.sh ${{ env.TESTING_CMD_PARAMS }} diagnostic
+
+      # Push the new TNF image to Quay.io.
+
+      - name: Authenticate against Quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          # Use a Robot Account to authenticate against Quay.io
+          # https://docs.quay.io/glossary/robot-accounts.html
+          username: ${{ secrets.QUAY_ROBOT_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+
+      - name: Push the newly built image to Quay.io
+        run: docker push --all-tags ${REGISTRY}/${IMAGE_NAME}
```

Comment:
Initially, pushing the image was handled by a dedicated job. 
GitHub makes it hard to persist resources between jobs, so the built image had to be `docker saved`, uploaded as an artifact, and then downloaded and `docker loaded` in the next job. This is really cumbersome, and adds unnecessary boilerplate to the workflow defintion.

An alternative approach would be to rebuild the image inside the push job, but building the TNF image takes about 5 minutes, so doing it twice increases the total execution time significantly. I also wanted to make sure that the image pushed to Quay.io is the exact one that has passed the testing.